Library/Homebrew/compat/md5.rb

```diff
@@ -1,28 +1,23 @@
 class Formula
   def self.md5(val)
-    stable.md5(val)
+    odie "MD5 support has been dropped for security reason"
   end
 end
 
 class SoftwareSpec
   def md5(val)
-    @resource.md5(val)
+    odie "MD5 support has been dropped for security reason"
   end
 end
 
 class Resource
   def md5(val)
-    @checksum = Checksum.new(:md5, val)
+    odie "MD5 support has been dropped for security reason"
   end
 end
 
 class Pathname
   def md5
-    require 'digest/md5'
-    opoo <<-EOS.undent
-    MD5 support is deprecated and will be removed in a future version.
-    Please switch this formula to #{Checksum::TYPES.map { |t| t.to_s.upcase } * ' or '}.
-    EOS
-    incremental_hash(Digest::MD5)
+    odie "MD5 support has been dropped for security reason"
```

Comment:
I think it's better to just error out. We do can add `Please switch this formula to #{Checksum::TYPES.map { |t| t.to_s.upcase } * ' or '}` into `odie` message.